src/main/java/com/puppycrawl/tools/checkstyle/checks/modifier/RedundantModifierCheck.java

```diff
@@ -73,6 +77,12 @@
  * automatically public, static and final just as their
  * annotation fields are automatically public and abstract.</p>
  *
+ * <p>A record class is implicitly final and cannot be abstract, these restrictions emphasize
+ * that the API of a record class is defined solely by its state description, and
+ * cannot be enhanced later by another class. Nested records are implicitly static. This avoids an
```

Comment:
>Nested records are implicitly static.

I do not see reflection of this in code.
Please make input for not nested record.
Please extend violation comments in Input to have message part that define what modifier is redundant.

Reply:
> > Nested records are implicitly static.
> 
> I do not see reflection of this in code. Please make input for not nested record. 

There is a top-level record in the example, top-level records can't have the keyword static (compile-time error), examples for nested records are there.

> Please extend violation comments in Input to have message part that define what modifier is redundant.

Added where there was only a single violation.